packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -182,7 +182,7 @@ async function getIslandContext (event: H3Event): Promise<NuxtIslandContext> {
   const context = event.method === 'GET' ? getQuery(event) : await readBody(event)
 
   const ctx: NuxtIslandContext = {
-    url: '/',
+    url: event.path,
```

Comment:
I don't think this is routeRules but rather vue router which is redirecting us, probably because we are still loading the router plugin for use within island.

I'm reluctant to use the `event.path` as this will still have the same issue with any middleware/redirects that apply to the original path that is loading the component.

Instead, I think we should add some more logic to the router plugin to check we're not in an island context, e.g.:

https://github.com/nuxt/nuxt/blob/17ff26ea931e3b430443ab28cc4ff62da205bb78/packages/nuxt/src/pages/runtime/plugins/router.ts#L146

Later we can perhaps opt-out entirely of routing (or, ideally, provide a 'stub' route rather than fully initialising vue-router). But we need to ensure that `useRoute()` continues to work within server components so we keep user-land differences to a minimum.

Reply:
Yes this would cause an issue if there's rules applied on `/__nuxt_island/**`

Is there a possibility to ignore everything for `/__nuxt_island` ? It implies removing probably both routeRules, nitro global middlewares and the router.

Currently working on server pages, we'll probably need to disable the router plugin and provide a stubbed route for `useRoute` because it currently doesn't work within islands

Reply:
Ah yes, that makes sense if Vue Router is handling it. I've pushed up a fix for that.

It's not really clear in this PR but the `event.path` will always be `/__nuxt_island/**`, I've added a comment to make that clearer. I have a strong feeling that we should go with this path (assuming there isn't a good reason for using `/`), it's arbitrary as the user should never interact with it but it does solve another issue and may avoid issues in the future.

For example, if we have a plugin that does something when the user visits the home page directly, it will always get triggered when an island renders, we can work around with using `env: { islands: false }`, but this seems like an easy way to trip up people.

```ts
export default defineNuxtPlugin({
  setup() {
    const route = useRoute()
    if (route.path === '/') {
      // This will run in an island component
    }
  }
})
```

It feels like we're plugging holes with the implementation so gives us more protection against other possible bugs.